libhb/qsv_common.c

```diff
@@ -28,6 +29,9 @@
 #include "libavutil/hwcontext_qsv.h"
 #include "libavutil/hwcontext.h"
 
+static HBQSVFramesContext *hb_dec_qsv_frames_ctx = NULL;
```

Comment:
More globals that are used during encoding...

I suggest setting pv->context->opaque = job in decavcodecvInit where get_format is assigned and then retrieve the job where these are used in hb_qsv_get_format (which calls hb_qsv_hw_frames_init which calls  hb_create_ffmpeg_pool), then remove function hb_qsv_update_frames_context